packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -635,6 +635,12 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
   if (rescheduleUid) {
     originalRescheduledBooking = await getOriginalRescheduledBooking(rescheduleUid);
   }
+  // If the user is not the owner of the event, new booking should be always pending.
+  // Otherwise, an owner rescheduling should be always accepted.
+  // Before comparing make sure that userId is set, otherwise undefined === undefined
+  const userReschedulingIsOwner = userId && originalRescheduledBooking?.user?.id === userId;
+  const isConfirmedByDefault =
+    (!eventType.requiresConfirmation && !stripeAppData.price) || userReschedulingIsOwner;
```

Comment:
Is there any way we can perhaps expand this into regular if-else? I think it's a bit difficult to read as it is